### PR TITLE
Chat feature

### PR DESCRIPTION
**Pull request by micosmin**
Implemented and designed
Redesigned posts page a bit
Need tests for messages

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,8 +8,9 @@ git_source(:github) do |repo_name|
 end
 
 gem 'faker'
-
+gem 'coffee-rails'
```
[the remark this answers is not in the document]
Comment (LaurenMullally) [reply Gemfile:10]: You're a picky hound...

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -13,3 +13,214 @@
  *= require_tree .
  *= require_self
  */
+
+.navbar {
```
Comment (m-budryte) [inline app/assets/stylesheets/application.css:17]: Good use of CSS, namings are clear therefore make it easy to change in the future

File: app/assets/javascripts/channels/room.coffee [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      html += '<div class = "msg-info">'
+      html += '<div class="message-user"><strong>'
+      html += data.first_name
+      html += ':</strong></div>'
+      html += '<div class="message-time">'
+      html += new Date().toLocaleString()
+      html += '</div>'
+      html += '</div>'
+      html += '<div class="message-content"><p>'
+      html += data.content
```
Comment (LaurenMullally) [inline app/assets/javascripts/channels/room.coffee:22]: I really like this format, It's super clear and easy to read.

File: app/assets/javascripts/channels/room.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  },
+
+  disconnected: function() {
+    // Called when the subscription has been terminated by the server
+  },
+
+  received: function(data) {
+    // Called when there's incoming data on the websocket for this channel
+  }
+});
```
Comment (LaurenMullally) [inline app/assets/javascripts/channels/room.js:13]: Do we need this file? Seems like a duplicate of the Coffee Script one?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -8,8 +8,7 @@ class ApplicationController < ActionController::Base
   protected
 
   def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:first_name,:last_name,:email,:password) }
-    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:first_name,:last_name,:email,:password) }
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:first_name, :last_name, :email, :password) }
```
[the remark this answers is not in the document]
Comment (m-budryte) [reply app/controllers/application_controller.rb:11]: consider using do...end

**Comment by LaurenMullally** [comment]
Good job Cosmin!

**Comment by micosmin** [comment]
Thanks Team! Was easy to implement due to all the hard work we all did last week and the research this week!

File: app/assets/javascripts/channels/room.coffee (hunk 3 above)
Comment (micosmin) [reply app/assets/javascripts/channels/room.coffee:22]: Indeed. Changed it to this as it was hard to work on the one from the example we used

File: app/assets/stylesheets/application.css (hunk 2 above)
Comment (micosmin) [reply app/assets/stylesheets/application.css:17]: I have a good CSS Flexbox tutorial to quickly learn and apply it in practice

File: app/controllers/application_controller.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (micosmin) [reply app/controllers/application_controller.rb:11]: I don't think you need a do in a def. Happy to change if mistaken.

**Pull request merged**